WordPress/src/test/java/org/wordpress/android/ui/reader/repository/usecases/GetDiscoverCardsUseCaseTest.kt

```diff
@@ -0,0 +1,102 @@
+import org.wordpress.android.datasets.wrappers.ReaderPostTableWrapper
+import org.wordpress.android.fluxc.utils.AppLogWrapper
+import org.wordpress.android.models.discover.ReaderDiscoverCard.InterestsYouMayLikeCard
+import org.wordpress.android.models.discover.ReaderDiscoverCard.ReaderPostCard
+import org.wordpress.android.test
+import org.wordpress.android.ui.reader.ReaderConstants
+
+@InternalCoroutinesApi
+@RunWith(MockitoJUnitRunner::class)
+class GetDiscoverCardsUseCaseTest {
+    private lateinit var useCase: GetDiscoverCardsUseCase
+    private val readerDiscoverCardsTableWrapper: ReaderDiscoverCardsTableWrapper = mock()
+    private val parseDiscoverCardsJsonUseCase: ParseDiscoverCardsJsonUseCase = mock()
+    private val mockedJsonArray: JSONArray = mock()
+    private val mockedJsonObject1: JSONObject = mock()
+    private val mockedJsonObject2: JSONObject = mock()
+    private val readerPostTableWrapper: ReaderPostTableWrapper = mock()
+    private val appLogWrapper: AppLogWrapper = mock()
+
+    @Before
+    fun setUp() {
+        useCase = GetDiscoverCardsUseCase(
+                parseDiscoverCardsJsonUseCase,
+                readerDiscoverCardsTableWrapper,
+                readerPostTableWrapper,
+                appLogWrapper,
+                TEST_DISPATCHER
+        )
+        whenever(parseDiscoverCardsJsonUseCase.convertListOfJsonArraysIntoSingleJsonArray(anyOrNull()))
+                .thenReturn(mockedJsonArray)
+        whenever(mockedJsonArray.length()).thenReturn(2)
+        whenever(mockedJsonArray.getJSONObject(0)).thenReturn(mockedJsonObject1)
+        whenever(mockedJsonArray.getJSONObject(1)).thenReturn(mockedJsonObject2)
+        whenever(readerDiscoverCardsTableWrapper.loadDiscoverCardsJsons()).thenReturn(listOf(""))
+        whenever(parseDiscoverCardsJsonUseCase.parseInterestCard(anyOrNull())).thenReturn(mock())
+        whenever(parseDiscoverCardsJsonUseCase.parseSimplifiedPostCard(anyOrNull())).thenReturn(Pair(101, 102))
+        whenever(readerPostTableWrapper.getBlogPost(anyLong(), anyLong(), anyBoolean())).thenReturn(mock())
+        whenever(mockedJsonObject1.getString(ReaderConstants.JSON_CARD_TYPE))
+                .thenReturn(ReaderConstants.JSON_CARD_POST)
+        whenever(mockedJsonObject2.getString(ReaderConstants.JSON_CARD_TYPE))
+                .thenReturn(ReaderConstants.JSON_CARD_INTERESTS_YOU_MAY_LIKE)
+    }
+
+    @Test
+    fun `interest you might like card is transformed into InterestsYouMayLikeCard object`() = test {
+        // Arrange
+        whenever(mockedJsonObject1.getString(ReaderConstants.JSON_CARD_TYPE))
+                .thenReturn(ReaderConstants.JSON_CARD_INTERESTS_YOU_MAY_LIKE)
+        // Act
+        val result = useCase.get()
+        // Assert
+        assertThat(result.cards[0]).isInstanceOf(InterestsYouMayLikeCard::class.java)
+    }
+
+    @Test
+    fun `post card is transformed into ReaderPostCard object`() = test {
```

Comment:
Same as above 😄

Reply:
Fixed in 66f7a9314c5d6d620387ca60ff34466f69fb0f42